File: InventoryAPI/InventoryAPI/Models/Item.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace InventoryAPI.Models
{
    public partial class Item
    {
        public Item()
        {
            Orders = new HashSet<Order>();
        }

        public long Id { get; set; }
        public string Category { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long? Price { get; set; }
        public long? Quantity { get; set; }
        public string Total { get; set; }
        public long? SupplierId { get; set; }

        public virtual Supplier Supplier { get; set; }
        public virtual ICollection<Order> Orders { get; set; }
    }
}

File: InventoryApp/src/app/models/item.ts
export class Item {
    id: number;
    category: string;
    name: string;
    description: string;
    price: number;
    quantity: number;
    total: number;
    supplierId: string;
    static fromJsonArray(jsonArray): Item[] {
      const items = [];
  
      jsonArray.forEach(element => {
        const item = new Item();
        item.id = element.id;
        item.category = element.category;
        item.description = element.name+" "+element.description;
        item.price = element.price;
        item.quantity = element.quantity;
        item.total = item.price*item.quantity;
        item.supplierId = element.supplierId;
        items.push(item);
      });
  
      return items;
    }
  }